CMakeLists.txt

```diff
@@ -97,6 +97,20 @@ set_option_if(
 set_option_if(ACTS_BUILD_PLUGIN_LEGACY ACTS_BUILD_EVERYTHING)
 set_option_if(ACTS_BUILD_PLUGIN_AUTODIFF ACTS_BUILD_EVERYTHING)
 
+# feature tests
+include(CheckCXXSourceCompiles)
+
+# function that tests if the root installation is compatible (i.e., compiled with C++17)
+function(check_root_compatibility)
+  get_target_property(ROOT_INCLUDE_DIR ROOT::Core INTERFACE_INCLUDE_DIRECTORIES)
+  set(CMAKE_REQUIRED_FLAGS -std=c++17)
+  set(CMAKE_REQUIRED_INCLUDES ${ROOT_INCLUDE_DIR})
```

Comment:
Hm, maybe one thing: do these affect the rest of our targets? If so, maybe you can reset them to the previous values after the check completes?

Reply:
As far es I understand, variables set in a CMake `function` environment are local to this function, so they do not pollute outer scopes (i.e., functions open a new scope, see https://cmake.org/cmake/help/latest/command/function.html)
I also quickly tested and verified that these values cannot be accessed from outside.

Reply:
Ok perfect.